### PR TITLE
Allow pull requests Manually Merged option to be used by non-admins

### PR DESCRIPTION
**Pull request by brechtvl**
Currently this feature is only available to admins, but there is no clear reason why. If a user can actually merge pull requests, then this seems fine as well.

This is useful in situations where direct pushes to the repository are commonly done by developers.

### CODE REVIEW

File: templates/repo/issue/view_content/pull.tmpl [hunk 1]
```diff
@@ -263,7 +263,7 @@
 								},
 								{
 									'name': 'manually-merged',
-									'allowed': {{and $prUnit.PullRequestsConfig.AllowManualMerge $.IsRepoAdmin}},
+									'allowed': {{and $prUnit.PullRequestsConfig.AllowManualMerge}},
```
Comment (delvh) [inline templates/repo/issue/view_content/pull.tmpl]:
```suggestion
									'allowed': {{$prUnit.PullRequestsConfig.AllowManualMerge}},
```
should be equivalent?

**Review by delvh — APPROVED** [review]

File: routers/web/repo/issue.go [hunk 2]
```diff
@@ -1927,7 +1927,7 @@ func ViewIssue(ctx *context.Context) {
 			if pull.CanAutoMerge() || pull.IsWorkInProgress(ctx) || pull.IsChecking() {
 				return false
 			}
-			if (ctx.Doer.IsAdmin || ctx.Repo.IsAdmin()) && prConfig.AllowManualMerge {
+			if ctx.Repo.CanWrite(unit.TypeCode) && prConfig.AllowManualMerge {
```
Comment (lunny) [inline routers/web/repo/issue.go]: This is not right. At least, we need to check two functions
`pull_service.IsUserAllowedToMerge(ctx, pull, perm, ctx.Doer)` and `pb.IsUserMergeWhitelisted`, if it's not admin.

**Review by lunny — CHANGES_REQUESTED** [review]
The change may bypass protected branch rules.

File: routers/web/repo/issue.go (hunk 2 above)
Comment (puni9869) [reply routers/web/repo/issue.go]: I agree with lunny on this

**Review by lunny — APPROVED** [review]

**Review by puni9869 — APPROVED** [review]

**Pull request merged**